### PR TITLE
Edge function router

### PR DESCRIPTION
**Pull request by jonator**
## What is the purpose of the change

@p0mvn requests we add our router to an edge function so that he can compare our routes against his during automated testing.

Example route:
`/api/router/token-out-given-in?tokenInDenom=uion&tokenInAmount=100&tokenOutDenom=uosmo`

IBC denoms will need to be URL encoded

### CODE REVIEW

File: packages/web/pages/api/router/token-out-given-in.ts [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+    initialAmount: string;
+    pools: {
+      id: string;
+    }[];
+    tokenOutDenoms: string[];
+    tokenInDenom: string;
+  }[];
+};
+
+export default async function routeTokenOutGivenIn(req: Request) {
```
Comment (p0mvn) [inline packages/web/pages/api/router/token-out-given-in.ts:33]: Do you mind also exposing the candidate route selection logic?

That is, for the given two denoms, return the candidate routes

File: packages/pools/src/router/routes.ts [hunk 2]
```diff
@@ -31,10 +31,6 @@ export type OptimizedRoutesParams = {
   /** **Ordered** IDs of pools to be prioritized in route selection.
    *  The first pools in the array will be selected first. */
   preferredPoolIds?: string[];
-  /** IDs of pools receiving OSMO incentives. */
-  incentivizedPoolIds: string[];
-  /** Min/base denom of stake currency of chain (OSMO) */
-  stakeCurrencyMinDenom: string;
```
Comment (p0mvn) [inline packages/pools/src/router/routes.ts:37]: Is this a drive-by change?

**Review by p0mvn — APPROVED** [review]
LGTM

File: packages/web/pages/api/router/token-out-given-in.ts (hunk 1 above)
Comment (jonator) [reply packages/web/pages/api/router/token-out-given-in.ts:33]: You want candidate routes returned in API?

File: packages/pools/src/router/routes.ts (hunk 2 above)
Comment (jonator) [reply packages/pools/src/router/routes.ts:37]: I forgot to remove it when I removed the OSMO 2 hop discount, as that's all it was used for. So yes you could say it's drive by.

File: packages/web/pages/api/router/token-out-given-in.ts (hunk 1 above)
Comment (p0mvn) [reply packages/web/pages/api/router/token-out-given-in.ts]: Yeah
